src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -110,6 +111,10 @@ public WorkspaceCloudContext getCloudContext(UUID workspaceId, AuthenticatedUser
   @Traced
   public String createGoogleContext(UUID workspaceId, AuthenticatedUserRequest userReq) {
     samService.workspaceAuthz(userReq, workspaceId, SamUtils.SAM_WORKSPACE_WRITE_ACTION);
+    Workspace workspace = workspaceDao.getWorkspace(workspaceId);
+    if (!WorkspaceStage.MC_WORKSPACE.equals(workspace.workspaceStage())) {
```

Comment:
Should this check be in `deleteGoogleContext` too? Right now it would just return success for a Rawls workspace which is fine, but slightly less informative than a `StageDisabledException`.

Reply:
Good point. done.